src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -39,16 +39,27 @@
     protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) throws ServletException, IOException {
         ModuleDataRetriever mdr = new ModuleDataRetriever(request.getResourceResolver());
         String searchParam = getParam(request, "search");
+        String countParam = getParam(request, "search");
         String keyParam = getParam(request, "key");
         String directionParam = getParam(request, "direction");
         String offset = getParam(request, "offset");
         String limit = getParam(request, "limit");
 
         try {
-            List<Map<String, Object>> payload = mdr.getModulesSort(searchParam, keyParam, directionParam, offset, limit);
             response.setContentType("application/json");
             Writer w = response.getWriter();
-            w.write(new ObjectMapper().writer().withDefaultPrettyPrinter().writeValueAsString(payload));
+            if(searchParam != ""){
```

Comment:
I'm not quite sure what's happening with this `if` statement, but it doesn't seem right. We shouldn't be only returning the row count depending on whether the user supplied a search parameter. Let's talk on Monday about how this ought to look.